dedupe/convenience.py

```diff
@@ -24,55 +24,88 @@ def consoleLabel(deduper): # pragma: no cover
+        n_distinct_in_buffer = sum(x[1]=='distinct' for x in record_label_buffer)
+                                                                                        
+        n_match, n_distinct = (len(deduper.training_pairs['match']) + n_match_in_buffer,
+                        len(deduper.training_pairs['distinct']) + n_distinct_in_buffer)
+      
+        label = ''
+
+        for pair in record_pair:
+            for field in fields:
+                line = "%s : %s" % (field, pair[field])
+                print(line, file=sys.stderr)
+            print(file=sys.stderr) 
+
+        print("{0}/10 positive, {1}/10 negative".format(n_match, n_distinct),
+                file=sys.stderr)
+        print('Do these records refer to the same thing?', file=sys.stderr)
+        valid_response = False
+        while not valid_response:
+            if len(record_label_buffer):
+                print('(y)es / (n)o / (u)nsure / (f)inished / (p)revious', file=sys.stderr)
+                label = input()
+                if label in ['y', 'n', 'u', 'f', 'p']:
+                    valid_response = True
+            else:
                 print('(y)es / (n)o / (u)nsure / (f)inished', file=sys.stderr)
                 label = input()
                 if label in ['y', 'n', 'u', 'f']:
                     valid_response = True
-
-            if label == 'y' :
-                examples['match'].append(record_pair)
-                labeled = True
-            elif label == 'n' :
-                examples['distinct'].append(record_pair)
-                labeled = True
-            elif label == 'f':
-                print('Finished labeling', file=sys.stderr)
-                finished = True
-            elif label != 'u':
-                print('Nonvalid response', file=sys.stderr)
-                raise
-
-        if labeled :
-            deduper.markPairs(examples)
+                if label == 'p':
+                    print('No record in memory: cannot use (p)revious', file=sys.stderr)
+
+        if label == 'y' :
+            record_label_buffer.insert(0, (record_pair, 'match'))
+            uncertain_pairs.pop() # Remove current pair form uncertain list
+        elif label == 'n' :
+            record_label_buffer.insert(0, (record_pair, 'distinct'))
+            uncertain_pairs.pop() 
+        elif label == 'f':
+            print('Finished labeling', file=sys.stderr)
+            finished = True
+        elif label == 'u':
+            record_label_buffer.insert(0, (record_pair, 'uncertain'))
+            uncertain_pairs.pop() 
+        elif (label == 'p') and len(record_label_buffer):
+            use_previous = True
+        else:
+            print('Nonvalid response', file=sys.stderr)
+            raise
         
+        if len(record_label_buffer) > buffer_len:
+            (record_pair, true_label) = record_label_buffer.pop()
+            if true_label in ['distinct', 'match']:
```

Comment:
I wanted to have distinct names for the user input (`y`, `n`, etc.) and the actual label (`match`,`distinct`)... I can change `label` to `user_input` and `true_label` to `label`. Would that work ?